README.md
# fake-gen

[![NPM](https://img.shields.io/npm/v/fake-gen.svg)](https://www.npmjs.com/package/fake-gen) [![code style: prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg)](https://github.com/prettier/prettier) [![Build Status](https://travis-ci.com/roggervalf/fake-gen.svg?branch=master)](https://travis-ci.com/roggervalf/fake-gen) [![NPM downloads](https://img.shields.io/npm/dm/fake-gen)](https://www.npmjs.com/package/fake-gen) [![Coverage Status](https://coveralls.io/repos/github/roggervalf/fake-gen/badge.svg?branch=master)](https://coveralls.io/github/roggervalf/fake-gen?branch=master) [![deno doc](https://doc.deno.land/badge.svg)](https://deno.land/x/fake_gen) [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)

## About

Faker data generator, progressive refactor from [faker](https://www.npmjs.com/package/faker)

## Install

```bash
npm install --save fake-gen
```

Or

```bash
yarn add fake-gen
```

## Deno import

```ts
// @deno-types="https://raw.githubusercontent.com/roggervalf/fake_gen/master/dist/main.d.ts"
import {
  DateTime,
  Internet,
  Random,
  Unique,
  Vehicle
} from 'https://raw.githubusercontent.com/roggervalf/fake_gen/master/dist/main.es.js';
```

or

```ts
// @deno-types="https://deno.land/x/fake_gen@master/dist/main.d.ts"
import {
  DateTime,
  Internet,
  Random,
  Unique,
  Vehicle
} from 'https://deno.land/x/fake_gen@master/dist/main.es.js';
```

## Node import

```js
import { DateTime, Internet, Random, Unique, Vehicle } from 'fake-gen';
```

**Example:**

```js
const { DateTime, Internet, Random, Unique, Vehicle } = require('fake-gen');

const dateTime = new DateTime();

console.log(dateTime.past());
// expects a random date instance between 1 year in the past and now

console.log(dateTime.future());
// expects a random date instance between 1 year in the future and now

const random = new Random();

console.log(random.alpha());
// expects a random character from 'a' to 'z'

console.log(random.alphaNumeric());
// expects a random character from 'a' to 'z' or '0' to '9'

console.log(random.arrayElement([1, 2, 3]));
// expects a random element from the provided array, i.e: 2

console.log(random.boolean());
// expects a random boolean, i.e: false

console.log(random.float());
// expects a random float number, i.e: 43173.42

console.log(random.hexadecimal());
// expects a random hexadecimal expression, i.e: 0xF

console.log(random.number(100));
// expects a random number from 0 to 100

console.log(random.uuid());
// expects a random uuid, i.e: 49e71c40-9b21-4371-9699-2def33f62e66

random.initSeed(100);
console.log(random.number(100));
// expects a 54

const internet = new Internet();

console.log(internet.avatar());
// expects a random avatar uri, i.e: https://randomuser.me/api/portraits/women/54.jpg

console.log(internet.ip());
// expects a random IP, i.e: 200.105.198.100

console.log(internet.ipv6());
// expects a random IPv6, i.e: 9cda:87cd:9617:370e:8d56:d698:19c8:c195

console.log(internet.protocol());
// expects a random ip, i.e: https

const vehicle = new Vehicle();

console.log(vehicle.vin());
// expects a random vehicle identification number, i.e: YV1MH682762184654

console.log(vehicle.vrm());
// expects a random vehicle registration mark, i.e: GL19AAQ

const unique = new Unique();

console.log(
  unique.execute({
    scope: 'scope',
    method: random.arrayElement,
    args: [[1, 2, 3]],
    model: random
  })
);
// expects a random unique value from arrayElement function, scoped by 'scope' string, i.e: 1
```

## Methods

- dateTime
  - future
  - past
- internet
  - avatar
  - ip
  - ipv6
  - protocol
- random
  - alpha
  - alphaNumeric
  - arrayElement
  - boolean
  - float
  - hexadecimal
  - number
  - uuid
- unique
  - clear
  - execute
- vehicle
  - vin
  - vrm

## Article

[How to build a Deno module](https://medium.com/@rogger.valverde/how-to-build-a-deno-module-dc383eee8edb)

## Contributing

Fork the repo, make some changes, submit a pull-request! Here is the [contributing](contributing.md) doc that has some details.

## License

MIT © [roggervalf](https://github.com/roggervalf)


CHANGELOG.md
# [1.11.0](https://github.com/roggervalf/fake-gen/compare/v1.10.0...v1.11.0) (2021-02-18)


### Features

* **datetime:** extending past and future methods by adding days option ([175c830](https://github.com/roggervalf/fake-gen/commit/175c8309cbb14a2bd55035e7fac5626b2289f45a))

# [1.10.0](https://github.com/roggervalf/fake-gen/compare/v1.9.0...v1.10.0) (2021-02-02)


### Features

* **internet:** adding ipv6 method to generate IPv6 addresses ([ab8c690](https://github.com/roggervalf/fake-gen/commit/ab8c6905a39c824645254bf5e249859026a778a9))

# [1.9.0](https://github.com/roggervalf/fake-gen/compare/v1.8.0...v1.9.0) (2021-01-18)


### Features

* **datetime:** adding DateTime module ([30b49cf](https://github.com/roggervalf/fake-gen/commit/30b49cf78f3ccadd2ba971d06f67588cea6933c1))

# [1.8.0](https://github.com/roggervalf/fake-gen/compare/v1.7.0...v1.8.0) (2021-01-11)


### Features

* **vehicle:** adding new module with vin and vrm methods ([4d50540](https://github.com/roggervalf/fake-gen/commit/4d50540990f1923bac9e3047b7c01dde315ef3d0))

# [1.7.0](https://github.com/roggervalf/fake-gen/compare/v1.6.0...v1.7.0) (2021-01-04)


### Features

* **unique:** new module to handle unique values by calling functions ([d7afdc1](https://github.com/roggervalf/fake-gen/commit/d7afdc10e84f6f57e685e2c3bbf9615e0a81d6f3))

# [1.6.0](https://github.com/roggervalf/fake-gen/compare/v1.5.0...v1.6.0) (2020-12-30)


### Features

* **random:** adding alphanumeric method ([5639f7b](https://github.com/roggervalf/fake-gen/commit/5639f7b9f6f45d5effc0fabb6dead8a85cf9cbf4))
* **random:** adding hexadecimal method ([bfca168](https://github.com/roggervalf/fake-gen/commit/bfca168e0d409ff62d3d5e7f5d686030dbe64594))

# [1.5.0](https://github.com/roggervalf/fake-gen/compare/v1.4.0...v1.5.0) (2020-12-29)


### Features

* **random:** adding alpha method ([6644cb8](https://github.com/roggervalf/fake-gen/commit/6644cb8345e452bf03607e31c90105899259905d))

# [1.4.0](https://github.com/roggervalf/fake-gen/compare/v1.3.0...v1.4.0) (2020-12-28)


### Features

* **internet:** adding ip method ([8739d9c](https://github.com/roggervalf/fake-gen/commit/8739d9cbdde3e35f1dcf5e454fb39cd78fa8715c))
* **internet:** adding protocol method ([b814cdd](https://github.com/roggervalf/fake-gen/commit/b814cdd5bdf1092941f5c3fdb0a6d202b8ab4a07))

# [1.3.0](https://github.com/roggervalf/fake-gen/compare/v1.2.0...v1.3.0) (2020-12-27)


### Features

* **internet:** adding new module with avatar method ([e20eca2](https://github.com/roggervalf/fake-gen/commit/e20eca25a9bb24c08c489fac488bad85fa43e025))

# [1.2.0](https://github.com/roggervalf/fake-gen/compare/v1.1.0...v1.2.0) (2020-12-26)


### Features

* **random:** adding arrayElement, float and boolean methods ([2ccae0d](https://github.com/roggervalf/fake-gen/commit/2ccae0de22ee9c240e0c97eb4a556b55bf85ddaf))

# [1.1.0](https://github.com/roggervalf/fake-gen/compare/v1.0.0...v1.1.0) (2020-12-24)


### Features

* **uuid:** adding uuid method to random module ([effb637](https://github.com/roggervalf/fake-gen/commit/effb637711a14c4b17649f98ee3a89eb9ad91421))

# 1.0.0 (2020-12-23)


### Features

* **random:** adding first module with number generator ([883cdea](https://github.com/roggervalf/fakerts/commit/883cdead148cf77319a9a73e5a776eac41562532))


src/modules/internet/definitions.ts
const avatarUris = [
  'https://randomuser.me/api/portraits/men/0.jpg',
  'https://randomuser.me/api/portraits/men/1.jpg',
  'https://randomuser.me/api/portraits/men/2.jpg',
  'https://randomuser.me/api/portraits/men/3.jpg',
  'https://randomuser.me/api/portraits/men/4.jpg',
  'https://randomuser.me/api/portraits/men/5.jpg',
  'https://randomuser.me/api/portraits/men/6.jpg',
  'https://randomuser.me/api/portraits/men/7.jpg',
  'https://randomuser.me/api/portraits/men/8.jpg',
  'https://randomuser.me/api/portraits/men/9.jpg',
  'https://randomuser.me/api/portraits/men/10.jpg',
  'https://randomuser.me/api/portraits/men/11.jpg',
  'https://randomuser.me/api/portraits/men/12.jpg',
  'https://randomuser.me/api/portraits/men/13.jpg',
  'https://randomuser.me/api/portraits/men/14.jpg',
  'https://randomuser.me/api/portraits/men/15.jpg',
  'https://randomuser.me/api/portraits/men/16.jpg',
  'https://randomuser.me/api/portraits/men/17.jpg',
  'https://randomuser.me/api/portraits/men/18.jpg',
  'https://randomuser.me/api/portraits/men/19.jpg',
  'https://randomuser.me/api/portraits/men/20.jpg',
  'https://randomuser.me/api/portraits/men/21.jpg',
  'https://randomuser.me/api/portraits/men/22.jpg',
  'https://randomuser.me/api/portraits/men/23.jpg',
  'https://randomuser.me/api/portraits/men/24.jpg',
  'https://randomuser.me/api/portraits/men/25.jpg',
  'https://randomuser.me/api/portraits/men/26.jpg',
  'https://randomuser.me/api/portraits/men/27.jpg',
  'https://randomuser.me/api/portraits/men/28.jpg',
  'https://randomuser.me/api/portraits/men/29.jpg',
  'https://randomuser.me/api/portraits/men/30.jpg',
  'https://randomuser.me/api/portraits/men/31.jpg',
  'https://randomuser.me/api/portraits/men/32.jpg',
  'https://randomuser.me/api/portraits/men/33.jpg',
  'https://randomuser.me/api/portraits/men/34.jpg',
  'https://randomuser.me/api/portraits/men/35.jpg',
  'https://randomuser.me/api/portraits/men/36.jpg',
  'https://randomuser.me/api/portraits/men/37.jpg',
  'https://randomuser.me/api/portraits/men/38.jpg',
  'https://randomuser.me/api/portraits/men/39.jpg',
  'https://randomuser.me/api/portraits/men/40.jpg',
  'https://randomuser.me/api/portraits/men/41.jpg',
  'https://randomuser.me/api/portraits/men/42.jpg',
  'https://randomuser.me/api/portraits/men/43.jpg',
  'https://randomuser.me/api/portraits/men/44.jpg',
  'https://randomuser.me/api/portraits/men/45.jpg',
  'https://randomuser.me/api/portraits/men/46.jpg',
  'https://randomuser.me/api/portraits/men/47.jpg',
  'https://randomuser.me/api/portraits/men/48.jpg',
  'https://randomuser.me/api/portraits/men/49.jpg',
  'https://randomuser.me/api/portraits/men/50.jpg',
  'https://randomuser.me/api/portraits/men/51.jpg',
  'https://randomuser.me/api/portraits/men/52.jpg',
  'https://randomuser.me/api/portraits/men/53.jpg',
  'https://randomuser.me/api/portraits/men/54.jpg',
  'https://randomuser.me/api/portraits/men/55.jpg',
  'https://randomuser.me/api/portraits/men/56.jpg',
  'https://randomuser.me/api/portraits/men/57.jpg',
  'https://randomuser.me/api/portraits/men/58.jpg',
  'https://randomuser.me/api/portraits/men/59.jpg',
  'https://randomuser.me/api/portraits/men/60.jpg',
  'https://randomuser.me/api/portraits/men/61.jpg',
  'https://randomuser.me/api/portraits/men/62.jpg',
  'https://randomuser.me/api/portraits/men/63.jpg',
  'https://randomuser.me/api/portraits/men/64.jpg',
  'https://randomuser.me/api/portraits/men/65.jpg',
  'https://randomuser.me/api/portraits/men/66.jpg',
  'https://randomuser.me/api/portraits/men/67.jpg',
  'https://randomuser.me/api/portraits/men/68.jpg',
  'https://randomuser.me/api/portraits/men/69.jpg',
  'https://randomuser.me/api/portraits/men/70.jpg',
  'https://randomuser.me/api/portraits/men/71.jpg',
  'https://randomuser.me/api/portraits/men/72.jpg',
  'https://randomuser.me/api/portraits/men/73.jpg',
  'https://randomuser.me/api/portraits/men/74.jpg',
  'https://randomuser.me/api/portraits/men/75.jpg',
  'https://randomuser.me/api/portraits/men/76.jpg',
  'https://randomuser.me/api/portraits/men/77.jpg',
  'https://randomuser.me/api/portraits/men/78.jpg',
  'https://randomuser.me/api/portraits/men/79.jpg',
  'https://randomuser.me/api/portraits/men/80.jpg',
  'https://randomuser.me/api/portraits/men/81.jpg',
  'https://randomuser.me/api/portraits/men/82.jpg',
  'https://randomuser.me/api/portraits/men/83.jpg',
  'https://randomuser.me/api/portraits/men/84.jpg',
  'https://randomuser.me/api/portraits/men/85.jpg',
  'https://randomuser.me/api/portraits/men/86.jpg',
  'https://randomuser.me/api/portraits/men/87.jpg',
  'https://randomuser.me/api/portraits/men/88.jpg',
  'https://randomuser.me/api/portraits/men/89.jpg',
  'https://randomuser.me/api/portraits/men/90.jpg',
  'https://randomuser.me/api/portraits/men/91.jpg',
  'https://randomuser.me/api/portraits/men/92.jpg',
  'https://randomuser.me/api/portraits/men/93.jpg',
  'https://randomuser.me/api/portraits/men/94.jpg',
  'https://randomuser.me/api/portraits/men/95.jpg',
  'https://randomuser.me/api/portraits/men/96.jpg',
  'https://randomuser.me/api/portraits/men/97.jpg',
  'https://randomuser.me/api/portraits/men/99.jpg',
  'https://randomuser.me/api/portraits/men/98.jpg',
  'https://randomuser.me/api/portraits/men/99.jpg',
  'https://randomuser.me/api/portraits/women/0.jpg',
  'https://randomuser.me/api/portraits/women/1.jpg',
  'https://randomuser.me/api/portraits/women/2.jpg',
  'https://randomuser.me/api/portraits/women/3.jpg',
  'https://randomuser.me/api/portraits/women/4.jpg',
  'https://randomuser.me/api/portraits/women/5.jpg',
  'https://randomuser.me/api/portraits/women/6.jpg',
  'https://randomuser.me/api/portraits/women/7.jpg',
  'https://randomuser.me/api/portraits/women/8.jpg',
  'https://randomuser.me/api/portraits/women/9.jpg',
  'https://randomuser.me/api/portraits/women/10.jpg',
  'https://randomuser.me/api/portraits/women/11.jpg',
  'https://randomuser.me/api/portraits/women/12.jpg',
  'https://randomuser.me/api/portraits/women/13.jpg',
  'https://randomuser.me/api/portraits/women/14.jpg',
  'https://randomuser.me/api/portraits/women/15.jpg',
  'https://randomuser.me/api/portraits/women/16.jpg',
  'https://randomuser.me/api/portraits/women/17.jpg',
  'https://randomuser.me/api/portraits/women/18.jpg',
  'https://randomuser.me/api/portraits/women/19.jpg',
  'https://randomuser.me/api/portraits/women/20.jpg',
  'https://randomuser.me/api/portraits/women/21.jpg',
  'https://randomuser.me/api/portraits/women/22.jpg',
  'https://randomuser.me/api/portraits/women/23.jpg',
  'https://randomuser.me/api/portraits/women/24.jpg',
  'https://randomuser.me/api/portraits/women/25.jpg',
  'https://randomuser.me/api/portraits/women/26.jpg',
  'https://randomuser.me/api/portraits/women/27.jpg',
  'https://randomuser.me/api/portraits/women/28.jpg',
  'https://randomuser.me/api/portraits/women/29.jpg',
  'https://randomuser.me/api/portraits/women/30.jpg',
  'https://randomuser.me/api/portraits/women/31.jpg',
  'https://randomuser.me/api/portraits/women/32.jpg',
  'https://randomuser.me/api/portraits/women/33.jpg',
  'https://randomuser.me/api/portraits/women/34.jpg',
  'https://randomuser.me/api/portraits/women/35.jpg',
  'https://randomuser.me/api/portraits/women/36.jpg',
  'https://randomuser.me/api/portraits/women/37.jpg',
  'https://randomuser.me/api/portraits/women/38.jpg',
  'https://randomuser.me/api/portraits/women/39.jpg',
  'https://randomuser.me/api/portraits/women/40.jpg',
  'https://randomuser.me/api/portraits/women/41.jpg',
  'https://randomuser.me/api/portraits/women/42.jpg',
  'https://randomuser.me/api/portraits/women/43.jpg',
  'https://randomuser.me/api/portraits/women/44.jpg',
  'https://randomuser.me/api/portraits/women/45.jpg',
  'https://randomuser.me/api/portraits/women/46.jpg',
  'https://randomuser.me/api/portraits/women/47.jpg',
  'https://randomuser.me/api/portraits/women/48.jpg',
  'https://randomuser.me/api/portraits/women/49.jpg',
  'https://randomuser.me/api/portraits/women/50.jpg',
  'https://randomuser.me/api/portraits/women/51.jpg',
  'https://randomuser.me/api/portraits/women/52.jpg',
  'https://randomuser.me/api/portraits/women/53.jpg',
  'https://randomuser.me/api/portraits/women/54.jpg',
  'https://randomuser.me/api/portraits/women/55.jpg',
  'https://randomuser.me/api/portraits/women/56.jpg',
  'https://randomuser.me/api/portraits/women/57.jpg',
  'https://randomuser.me/api/portraits/women/58.jpg',
  'https://randomuser.me/api/portraits/women/59.jpg',
  'https://randomuser.me/api/portraits/women/60.jpg',
  'https://randomuser.me/api/portraits/women/61.jpg',
  'https://randomuser.me/api/portraits/women/62.jpg',
  'https://randomuser.me/api/portraits/women/63.jpg',
  'https://randomuser.me/api/portraits/women/64.jpg',
  'https://randomuser.me/api/portraits/women/65.jpg',
  'https://randomuser.me/api/portraits/women/66.jpg',
  'https://randomuser.me/api/portraits/women/67.jpg',
  'https://randomuser.me/api/portraits/women/68.jpg',
  'https://randomuser.me/api/portraits/women/69.jpg',
  'https://randomuser.me/api/portraits/women/70.jpg',
  'https://randomuser.me/api/portraits/women/71.jpg',
  'https://randomuser.me/api/portraits/women/72.jpg',
  'https://randomuser.me/api/portraits/women/73.jpg',
  'https://randomuser.me/api/portraits/women/74.jpg',
  'https://randomuser.me/api/portraits/women/75.jpg',
  'https://randomuser.me/api/portraits/women/76.jpg',
  'https://randomuser.me/api/portraits/women/77.jpg',
  'https://randomuser.me/api/portraits/women/78.jpg',
  'https://randomuser.me/api/portraits/women/79.jpg',
  'https://randomuser.me/api/portraits/women/80.jpg',
  'https://randomuser.me/api/portraits/women/81.jpg',
  'https://randomuser.me/api/portraits/women/82.jpg',
  'https://randomuser.me/api/portraits/women/83.jpg',
  'https://randomuser.me/api/portraits/women/84.jpg',
  'https://randomuser.me/api/portraits/women/85.jpg',
  'https://randomuser.me/api/portraits/women/86.jpg',
  'https://randomuser.me/api/portraits/women/87.jpg',
  'https://randomuser.me/api/portraits/women/88.jpg',
  'https://randomuser.me/api/portraits/women/89.jpg',
  'https://randomuser.me/api/portraits/women/90.jpg',
  'https://randomuser.me/api/portraits/women/91.jpg',
  'https://randomuser.me/api/portraits/women/92.jpg',
  'https://randomuser.me/api/portraits/women/93.jpg',
  'https://randomuser.me/api/portraits/women/94.jpg',
  'https://randomuser.me/api/portraits/women/95.jpg',
  'https://randomuser.me/api/portraits/women/96.jpg',
  'https://randomuser.me/api/portraits/women/97.jpg',
  'https://randomuser.me/api/portraits/women/98.jpg',
  'https://randomuser.me/api/portraits/women/99.jpg'
];

export { avatarUris };
